calyx/src/passes/compile_ref.rs

```diff
@@ -14,7 +14,8 @@ use std::rc::Rc;
 /// 3. Remove all the ref cell mappings from the invoke statement
 /// 4. Inline all the mappings of ports to the invoke signature
 pub struct CompileRef {
-    port_names: HashMap<ir::Id, HashMap<ir::Id, HashMap<ir::Id, ir::Id>>>,
+    port_names:
+        HashMap<ir::Id, HashMap<ir::Id, HashMap<ir::Id, RRC<ir::Port>>>>,
```

Comment:
I'm not a huge fan of a `HashMap<HashMap<HashMap<...>>>`. Can we convince ourselves that this is needed here? If it is, then we need to update the documentation and make sure we know what each level of `HashMap` tracks